Controller/SubscriptionWallChecker.php

```diff
@@ -80,7 +80,58 @@ public function isBlocked(ContentInterface $article)
      */
     public function isLoggedIn()
     {
-        return $this->authorizationChecker->isGranted('IS_AUTHENTICATED_FULLY');
+        return $this->ac->isGranted('IS_AUTHENTICATED_FULLY');
     }
 
+    /**
+     * @return mixed
+     */
+    public function hasProperSubscription()
+    {
+        if ($this->isLoggedIn()) {
+            $relationId = $this->ts->getToken()->getUser()->getRelation()->getId();
+            $subscribedSubscriptions = $this->em
+                ->getRepository('Integrated\Bundle\SubscriptionBundle\Model\Subscription')
+                ->findByRelation($relationId);
+
+            $subscribedWalls = [];
+            foreach ($subscribedSubscriptions as $subscription) {
+                $wallsOfOneSubscription = $subscription->getType()->getSubscriptionWalls();
+
+                foreach ($wallsOfOneSubscription as $wall) {
+                    $subscribedWalls[] = $wall->getId();
+                }
+            }
+
+            $wallsThatBlockArticle = $this->getWallsThatBlockArticle();
+
+            foreach ($wallsThatBlockArticle as $wallThatBlocksArticle) {
+                if (in_array($wallThatBlocksArticle->getId(), $subscribedWalls)) {
+                    return true;
+                }
+            };
+            return false;
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * @return array
+     * @throws \Doctrine\DBAL\DBALException
+     */
+    public function getWallsThatBlockArticle()
+    {
+        $channel = $this->channel->getChannel();
+
+        $q = $this->em->getRepository('Integrated\Bundle\SubscriptionBundle\Model\SubscriptionWall')
```

Comment:
I've given the wrong example, can you try \Integrated\Bundle\SubscriptionBundle\Model\SubscriptionWall:class please? This makes it more robust when refactoring and strong typed 👍 .